sdk/data/azcosmos/cosmos_client.go

```diff
@@ -481,3 +482,79 @@ type pipelineRequestOptions struct {
 	isRidBased            bool
 	isWriteOperation      bool
 }
+
+func getAllowedHeaders() []string {
+	return []string{
+		cosmosHeaderRequestCharge,
```

Comment:
Proposed list:

```
cosmosHeaderRequestCharge,
cosmosHeaderActivityId,
cosmosHeaderEtag,
cosmosHeaderSessionToken,
cosmosHeaderConsistencyLevel,
cosmosHeaderQuery,
cosmosHeaderMaxItemCount,
cosmosHeaderQueryMetrics,
cosmosHeaderCorrelatedActivityId,
cosmosHeaderIsBatchRequest,
cosmosHeaderIsBatchAtomic,
cosmosHeaderIsBatchOrdered,
headerXmsDate,
headerContentType,
headerXmsVersion,
"x-ms-cosmos-llsn",
"x-ms-cosmos-item-llsn",
"x-ms-item-lsn",
"x-ms-cosmos-quorum-acked-llsn",
"x-ms-global-committed-lsn",
"x-ms-last-state-change-utc",
"x-ms-quorum-acked-lsn",
"x-ms-request-duration-ms",
"x-ms-transport-request-id",
"x-ms-documentdb-partitionkeyrangeid",
"x-ms-cosmos-physical-partition-id",
"x-ms-item-count",
"x-ms-start-epk",
"x-ms-end-epk",
"a-im",
```

Reply:
blacklisting model within the pipeline would depend on Core unless we make our own policy like we mentioned yesterday

Reply:
@kirankumarkolli When we discussed this, the point here is ROI. In order to support blacklisting either the base core SDK needs to do it or we need to build our own logging policy (non-trivial).

Given the lack of any sort of feedback loop and the unknown on when or if the base core SDK will support it, shouldn't we move forward with this?

Reply:
we discussed this offline and it's ok for the initial diagnostics work